gh_oidc_role/main.tf

```diff
@@ -11,29 +11,46 @@ data "tls_certificate" "thumprint" {
 }
 
 resource "aws_iam_role" "this" {
-  for_each = { for r in var.roles : r.name => r }
-
-  name = each.value.name
-  assume_role_policy = jsonencode({
-    Version = "2012-10-17"
-
-    Statement = {
-      Effect = "Allow"
-      Action = ["sts:AssumeRoleWithWebIdentity"]
-      Principal = {
-        Federated = aws_iam_openid_connect_provider.github.arn
-      }
-      Condition = {
-        StringLike = {
-          "token.actions.githubusercontent.com:sub" = "repo:${var.org_name}/${each.value.repo_name}:${each.value.claim}"
-        }
-      }
+  count = length(var.roles)
 
-    }
-  })
-  tags = local.common_tags
+
+  name               = tolist(var.roles)[count.index].name
+  assume_role_policy = data.aws_iam_policy_document.assume_role_policy[count.index].json
+  tags               = local.common_tags
 }
 
+data "aws_iam_policy_document" "assume_role_policy" {
+  count   = length(data.aws_iam_policy_document.oidc_assume_role_policy.*)
+  version = "2012-10-17"
```

Comment:
You can drop the version - the `aws_iam_policy_document` `json` attribute adds this in for us.